enumerables.rb

```diff
@@ -0,0 +1,220 @@
+            end
+          elsif arg.class == Class
+            my_each do |item|
+              return false if item.is_a?(arg) == false
+            end
+          elsif arg.nil?
+            my_each do |item|
+              return false unless item
+            end
+          else
+            my_each do |item|
+              return false if item != arg
+            end
+          end
+    end
+    true
+  end
+
+  # my_any?
+  def my_any?(arg = nil)
+    if block_given? # con bloque sin argumento
+      my_each do |item|
+        return true if yield(item) == true
+      end
+      # con argumento
+    elsif if arg.class == Regexp
+            my_each do |item|
+              return true if arg.match?(item) == true
+            end
+          elsif arg.class == Class
+            my_each do |item|
+              return true if item.is_a?(arg) == true
+            end
+          elsif arg.nil?
+            my_each do |item|
+              return true if item
+            end
+          else
+            my_each do |item|
+              return true if item == arg
+            end
+          end
+    end
+    false
+  end
+
+  # my_none?
+  def my_none?(arg = nil)
+    if block_given? # con bloque sin argumento
+      my_each do |item|
+        return false if yield(item) == true
+      end
+      # con argumento
+    elsif if arg.class == Regexp
+            my_each do |item|
+              return false if arg.match?(item) == true
+            end
+          elsif arg.class == Class
+            my_each do |item|
+              return false if item.is_a?(arg) == true
+            end
+          elsif arg.nil?
+            my_each do |item|
+              return false if item
+            end
+          else
+            my_each do |item|
+              return false if item == arg
+            end
+          end
+    end
+    true
+  end
+
+  def my_count(argument = nil)
+    count = 0
+    Array(self).my_each do |item|
+      if !block_given?
+        return Array(self).length if argument.nil?
+        next if item != argument
+
+        count += 1
+      elsif yield(item) == true
+        count += 1
+      end
+    end
+    count
+  end
+
+  # my_map
+
+  def my_map(argument = nil)
+    arr = []
+
+    if argument.class == Proc
+      Array(self).my_each do |item|
+        arr.push(argument.call(item))
+      end
+      return arr
+    end
+
+    return to_enum unless block_given?
+
+    Array(self).my_each do |item|
+      arr.push(yield(item))
+    end
+    arr
+  end
+
+  def my_inject(args = nil, sym = nil)
+    if (!args.nil? && sym.nil?) && (args.is_a?(Symbol) || args.is_a?(String))
+      sym = args
+      args = nil
+    end
+    if !block_given? && !sym.nil?
+      to_a.my_each do |index|
+        args = args.nil? ? index : args.send(sym, index)
+      end
+    else
+      to_a.my_each do |_item|
+        args = args.nil? ? index : yield(args, index)
+      end
+    end
+    args
+  end
+end
```

Comment:
# Failures
#my_inject
1. It is throwing an error, read details below :point_down: 
When I ran this statement `p ["dog", "door", "rod", "blade"].my_inject{ |memo, word| memo.length > word.length ? memo : word } `

the following error is thrown
```bash
Traceback (most recent call last):
        6: from enumerables.rb:222:in `<main>'
        5: from enumerables.rb:170:in `my_inject'
        4: from enumerables.rb:9:in `my_each'
        3: from enumerables.rb:9:in `times'
        2: from enumerables.rb:10:in `block in my_each'
        1: from enumerables.rb:171:in `block in my_inject'
enumerables.rb:222:in `block in <main>': undefined method `length' for #<Enumerator: ["dog", "door", "rod", "blade"]:index> (NoMethodError)
```

2.  It is throwing an error, read details below :point_down: 
When I ran this statement `p (0...50).my_inject{ |prod, n| prod * n }` and `p (0...50).my_inject(4){ |prod, n| prod * n }`

```bash
Traceback (most recent call last):
        5: from enumerables.rb:222:in `<main>'
        4: from enumerables.rb:170:in `my_inject'
        3: from enumerables.rb:9:in `my_each'
        2: from enumerables.rb:9:in `times'
        1: from enumerables.rb:10:in `block in my_each'
enumerables.rb:171:in `block in my_inject': undefined local variable or method `index' for 0...50:Range (NameError)
```

2. when a block is given without an initail value combines all elements of enum by applying a binary operation, specified by a block